packages/shared/src/hooks/useBookmarkProvider.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import { feature } from '../lib/featureManagement';
+import { useActiveFeedNameContext } from '../contexts';
+import { SharedFeedPage } from '../components/utilities';
+import { useConditionalFeature } from './useConditionalFeature';
+
+interface UseBookmarkProviderReturn {
+  highlightBookmarkedPost: boolean;
+}
+
+interface UseBookmarkProviderProps {
+  bookmarked?: boolean;
+}
+
+const useBookmarkProvider = ({
+  bookmarked = false,
+}: UseBookmarkProviderProps): UseBookmarkProviderReturn => {
+  const [isBookmarked] = useState<boolean>(bookmarked);
```

Comment:
We can probably turn this into a `useRef` as we don't expect state updates and we just want to hold the value here.